apps/evm/src/clients/api/queries/getImportablePositions/getImportableAaveSupplyPositions/index.ts

```diff
@@ -0,0 +1,142 @@
+}
+
+export interface GetImportableAaveSupplyPositionsOutput {
+  importableSupplyPositions: ImportableAaveSupplyPosition[];
+}
+
+export const getImportableAaveSupplyPositions = async ({
+  publicClient,
+  accountAddress,
+  aaveUiPoolDataProviderContractAddress,
+  aavePoolAddressesProviderContractAddress,
+}: GetImportableAaveSupplyPositionsInput): Promise<GetImportableAaveSupplyPositionsOutput> => {
+  const [userReservesData, reservesData] = await publicClient.multicall({
+    contracts: [
+      {
+        abi: aaveUiPoolDataProviderAbi,
+        address: aaveUiPoolDataProviderContractAddress,
+        functionName: 'getUserReservesData',
+        args: [aavePoolAddressesProviderContractAddress, accountAddress],
+      },
+      {
+        abi: aaveUiPoolDataProviderAbi,
+        address: aaveUiPoolDataProviderContractAddress,
+        functionName: 'getReservesData',
+        args: [aavePoolAddressesProviderContractAddress],
+      },
+    ],
+  });
+
+  if (userReservesData.status === 'failure' || reservesData.status === 'failure') {
+    throw new VError({ type: 'unexpected', code: 'somethingWentWrong' });
+  }
+
+  // Map user aToken balances by underlying token address
+  const userATokenBalanceMapping: {
+    [tokenAddress: Address]: {
+      supplyBalanceMantissa: bigint;
+    };
+  } = {};
+
+  let userHasBorrowPosition = false;
+
+  userReservesData.result.forEach(data => {
+    if (typeof data === 'number') {
+      return;
+    }
+
+    data.forEach(({ underlyingAsset, scaledATokenBalance, scaledVariableDebt }) => {
+      if (scaledVariableDebt > 0n) {
+        userHasBorrowPosition = true;
+        return;
+      }
+
+      userATokenBalanceMapping[underlyingAsset.toLowerCase() as Address] = {
+        supplyBalanceMantissa: scaledATokenBalance,
+      };
+    });
+  });
+
+  // Return no importable positions if user has any borrow position
+  if (userHasBorrowPosition) {
+    return {
+      importableSupplyPositions: [],
+    };
+  }
+
+  const importableSupplyPositions: ImportableAaveSupplyPosition[] = [];
+
+  // Map reserves by underlying token address
+  reservesData.result.forEach(data => {
+    if ('marketReferenceCurrencyPriceInUsd' in data) {
+      return;
+    }
+
+    data.forEach(
+      ({ liquidityRate, liquidityIndex, underlyingAsset, aTokenAddress, availableLiquidity }) => {
+        const tokenAddress = underlyingAsset.toLowerCase() as Address;
+
+        // Skip assets for which user has no supply position
+        if (!Object.prototype.hasOwnProperty.call(userATokenBalanceMapping, tokenAddress)) {
+          return;
+        }
+
+        const userABalances = userATokenBalanceMapping[tokenAddress];
+
+        if (userABalances.supplyBalanceMantissa === 0n) {
+          return;
+        }
+
+        const userSupplyBalanceMantissa = BigInt(
+          new BigNumber(Number(userABalances.supplyBalanceMantissa))
+            .multipliedBy(Number(liquidityIndex))
+            .dividedBy(Number(RAY))
```

Comment:
Is it possible to cause any precision loss by converting these values to numbers? Would it be safer to pass them as strings `bigInt.toString()`?